ruby/sinatra/app.rb

```diff
@@ -0,0 +1,77 @@
+
+require 'bundler/setup'
+
+require 'connection_pool'
+require 'date'
+require 'json'
+require 'redis'
+require 'sinatra'
+require 'sinatra/json'
+
+require 'ddtrace'
+require 'ddtrace/contrib/sinatra/tracer'
+
+require './post.rb'
+
+class App < Sinatra::Application
+  REDIS_HOST = '127.0.0.1'.freeze()
+  REDIS_PORT = 6379
+
+  configure do
+    settings.datadog_tracer.configure(default_service: 'sinatra-demo',
+                                      debug: true)
+
+    pool = ConnectionPool.new(size: 10) do
+      Redis.new(host: REDIS_HOST, port: REDIS_PORT)
+    end
+    set :redis_pool, pool
+  end
+
+  helpers do
+    def with_redis_conn(&block)
+      settings.redis_pool.with do |conn|
+        yield conn
+      end
+    end
+  end
+
+  get '/' do
+    erb :posts
+  end
+
+  get '/api/posts', provides: :json do
+    posts = []
+    with_redis_conn do |conn|
+      posts = Post.load_all(conn)
+    end
+
+    post_data = posts.map {|post| post.marshal()}
+    json post_data
+  end
+
+  get '/api/posts', provides: :html do
+    posts = []
+    with_redis_conn do |conn|
+      posts = Post.load_all(conn)
+    end
+
+    erb :posts_fragment, layout: nil, locals: {posts: posts}
+  end
+
+  post '/api/posts' do
```

Comment:
if the app should be deployed somewhere for testing purpose, this will be an endpoint that can fill a database in minutes with ad-hoc bot. Or we fill the database with preloaded data (so the POST endpoint is not required), or we use a simple authentication like in Rails example.